### DISCUSSION TITLE
What do I do with divergent change-IDs?

### DISCUSSION THREAD

**Discussion by BatmanAoD** [jj-vcs/jj#2485] [Q&A] [answered]
I generally keep my merge commits separate from any other commits. I merged `trunk` into a feature-branch, then made some changes without `jj new`. I only later realized that my working copy was still set to the merge commit, so I was editing that.

I hadn't noticed the "accidentally amended the working copy" advice in the FAQ, and I happened to push to my remote immediately after merging, so I attempted to resolve the issue by rebasing the modified commit onto the pushed commit: `jj rebase -r <merge+modifications> -d <branch>@origin`. Of course, these shared the same change-ID. So now my change-ID is `divergent`:

```
◉  omwkzvop?? kstrand@rigetti.com 2023-10-31 14:51:41.000 -06:00 <branch> HEAD@git 8824d66d
│  <description of post-merge changes>
◉    omwkzvop?? kstrand@rigetti.com 2023-10-31 13:32:06.000 -06:00 411fd733
├─╮  (empty) Merge '<trunk>' into '<branch>'

...
```

Is there a way to forcibly assign a new change-ID to one or the other of these? It may not matter, since it seems I can use the commit IDs for all the operations I'd want, but it feels like a strange state to be stuck in.

**Comment by ilyagr** [comment]
The "default" way to keep one of the different commits  is to `jj
duplicate` it.

**Comment by martinvonz** [comment, marked as answer]
As Ilya said, you can use `jj duplicate` to fix it. That will create a new commit with a different change id. You can then abandon the old commit (`8824d66d` in your case).

Incidentally, the pending PR #2477 adds an FAQ entry about this. We also have #555 about warning immediately when the divergence happens.
